src/localization.rs

```diff
@@ -0,0 +1,277 @@
+    ($fd:expr, $output:expr) => {{
+        let _ = $crate::wutil::unescape_and_write_to_fd($output, $fd);
+    }};
+}
+pub use fprint;
+
+#[macro_export]
+macro_rules! localized_fprint {
+    ($fd:expr, $id:expr, $args:expr) => {
+        $crate::localization::fprint!($fd, fish_fluent::localize!($id, $args))
+    };
+    ($fd:expr, $id:expr $(, $key:ident = $value:expr)* $(,)?) => {
+        $crate::localization::fprint!($fd, fish_fluent::localize!($id $(, $key = $value)*))
+    };
+}
+pub use localized_fprint;
+
+#[macro_export]
+macro_rules! localized_fprintln {
+    ($fd:expr, $id:expr, $args:expr) => {{
+        $crate::localization::localized_fprint!($fd, $id, $args);
+        $crate::wutil::write_newline_to_fd($fd);
+    }};
+    ($fd:expr, $id:expr $(, $key:ident = $value:expr)* $(,)?) => {{
+        $crate::localization::localized_fprint!($fd, $id $(, $key = $value)*);
+        $crate::wutil::write_newline_to_fd($fd);
+    }};
+}
+pub use localized_fprintln;
+
+#[macro_export]
+macro_rules! localized_print {
+    ($id:expr, $args:expr) => {
+        $crate::localization::localized_fprint!(libc::STDOUT_FILENO, $id, $args)
+    };
+    ($id:expr $(, $key:ident = $value:expr)* $(,)?) => {
+        $crate::localization::localized_fprint!(libc::STDOUT_FILENO, $id $(, $key = $value)*)
+    };
+}
+pub use localized_print;
+
+#[macro_export]
+macro_rules! localized_println {
+    ($id:expr, $args:expr) => {{
+        $crate::localization::localized_fprintln!(libc::STDOUT_FILENO, $id, $args);
+    }};
+    ($id:expr $(,$key:ident = $value:expr)* $(,)?) => {{
+        $crate::localization::localized_fprintln!(libc::STDOUT_FILENO, $id $(, $key = $value)*);
+    }};
+}
+pub use localized_println;
+
+#[macro_export]
+macro_rules! localized_eprint {
+    ($id:expr, $args:expr) => {
+        $crate::localization::localized_fprint!(libc::STDERR_FILENO, $id, $args)
+    };
+    ($id:expr $(, $key:ident = $value:expr)* $(,)?) => {
+        $crate::localization::localized_fprint!(libc::STDERR_FILENO, $id $(, $key = $value)*)
+    };
+}
+pub use localized_eprint;
+
+#[macro_export]
+macro_rules! localized_eprintln {
+    ($id:expr, $args:expr) => {{
+        $crate::localization::localized_fprintln!(libc::STDERR_FILENO, $id, $args);
+    }};
+    ($id:expr $(, $key:ident = $value:expr)* $(,)?) => {{
+        $crate::localization::localized_fprintln!(libc::STDERR_FILENO, $id $(, $key = $value)*);
+    }};
+}
+pub use localized_eprintln;
+
+#[macro_export]
+macro_rules! localized_sprint {
+    ($id:expr, $args:expr) => {
+        fish_fluent::localize!($id, $args)
+    };
+    ($id:expr $(, $key:ident = $value:expr)* $(,)?) => {
+        fish_fluent::localize!($id $(, $key = $value)*)
+    };
+}
+pub use localized_sprint;
+
+#[macro_export]
+macro_rules! sprintln {
+    ($input:expr) => {{
+        let mut res = $input.into_owned();
+        res.push('\n');
+        res
+    }};
+}
+pub use sprintln;
+
+#[macro_export]
+macro_rules! localized_sprintln {
```

Comment:
The "sprint" style naming is kinda consistent with fprint/eprint etc. but it is an anachronism from C's printf.
Maybe we should call these Rust-style `localized_format` and `localized_formatln`
(we'd only need to get used to `formatln`).
Not sure what other users of fluent do; we can also change it later..